sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/e-cert-cancellation-response/_tests_/ecert-cancellation-response-integration.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,425 @@
+          ]),
+        ).toBe(true);
+        // The file is not expected to be archived on SFTP.
+        expect(sftpClientMock.rename).not.toHaveBeenCalled();
+      },
+    );
+
+    it(
+      "Should cancel the e-certs when the Part-time e-cert cancellation response file" +
+        " has 2 detail records with valid document numbers.",
+      async () => {
+        // Arrange
+        // Create fake applications with disbursements to be cancelled.
+        const [, , firstDocumentNumber, secondDocumentNumber] =
+          SHARED_DOCUMENT_NUMBERS;
+        const firstApplicationPromise = saveFakeApplicationDisbursements(
+          db.dataSource,
+          undefined,
+          {
+            offeringIntensity: OfferingIntensity.partTime,
+            applicationStatus: ApplicationStatus.Completed,
+            firstDisbursementInitialValues: {
+              coeStatus: COEStatus.completed,
+              disbursementScheduleStatus: DisbursementScheduleStatus.Sent,
+              documentNumber: firstDocumentNumber,
+            },
+          },
+        );
+        const secondApplicationPromise = saveFakeApplicationDisbursements(
+          db.dataSource,
+          undefined,
+          {
+            offeringIntensity: OfferingIntensity.partTime,
+            applicationStatus: ApplicationStatus.Completed,
+            firstDisbursementInitialValues: {
+              coeStatus: COEStatus.completed,
+              disbursementScheduleStatus: DisbursementScheduleStatus.Sent,
+              documentNumber: secondDocumentNumber,
+            },
+          },
+        );
+        const [firstApplication, secondApplication] = await Promise.all([
+          firstApplicationPromise,
+          secondApplicationPromise,
+        ]);
+        const [firstDisbursement] =
+          firstApplication.currentAssessment.disbursementSchedules;
+        const [secondDisbursement] =
+          secondApplication.currentAssessment.disbursementSchedules;
+        mockDownloadFiles(sftpClientMock, [
+          PART_TIME_CANCELLATION_RESPONSE_FILE,
+        ]);
+        // Queued job.
+        const mockedJob = mockBullJob<void>();
+
+        // Act
+        const result = await processor.processQueue(mockedJob.job);
+
+        // Assert
+        expect(result).toStrictEqual([
+          "Process finalized with success.",
+          "Received cancellation files: 1.",
+        ]);
+        const downloadedFile = path.join(
+          process.env.ESDC_RESPONSE_FOLDER,
+          PART_TIME_CANCELLATION_RESPONSE_FILE,
+        );
+        // Check for the log messages.
+        expect(
+          mockedJob.containLogMessages([
+            "Received 1 e-cert cancellation response file(s) to process.",
+            `The downloaded file ${downloadedFile} contains 2 detail records.`,
+            `E-Cert with document number ${firstDocumentNumber} has been cancelled.`,
```

Comment:
Lines 245 and 246 are duplicated.

Reply:
The second line is suppose to check, secondDocumentNumber